### PR TITLE
[DEVOPS-194]  Fix absence of nix-prefetch-scripts in the iohk-ops runtime deps

### PR DESCRIPTION
**Pull request by deepfire**

### CODE REVIEW

**Comment by domenkozar** [comment]
This doesn't allow `iohk-ops` to be used outside of nix-shell on deployer machines, which is still inconvenient.

**Comment by deepfire** [comment]
https://github.com/jml/nix-haskell-example looks appropriate.

**Comment by domenkozar** [comment]
Since we're invoking shell, we can just call `nix-shell`, is there something preventing us that?

**Comment by deepfire** [comment]
So, it seems that what is needed is a wrapper around `iohk-ops`, that would extend `PATH`.

**Comment by deepfire** [comment]
@domenkozar, I don't follow -- the problem we are trying to solve is -- we need the `iohk-ops` binary to have access to `nix-prefetch-git` outside `nix-shell`.

Since it's a compiled binary, I don't understand how `nix-shell` can even enter the picture..

**Comment by domenkozar** [comment]
```
$ grep -R nix-prefetch iohk/
iohk/NixOps.hs:-- | The output of 'nix-prefetch-git'
iohk/iohk-ops.hs:  spec <- incmd o "nix-prefetch-git" [fromURL Ops.cardanoSlURL, fromCommit rev]
iohk/iohk-ops.hs:  spec <- incmd o "nix-prefetch-git" [fromURL Ops.cardanoSlExplorerURL, fromCommit rev]
iohk/iohk-ops.hs:  spec <- incmd o "nix-prefetch-git" [fromURL Ops.stack2NixURL, fromCommit rev]
```

These should be changed to invoke `nix-shell -p nix-prefetch-scripts --run "nix-prefetch-git ..."`

**Comment by deepfire** [comment]
Oh, sorry, I'm feeling a bit dumb now : -)

**Comment by domenkozar** [comment]
This was written in the issue description btw :)

**Comment by deepfire** [comment]
@domenkozar, now we have two options, though:

1. make a PATH-extending wrapper (a standard `nixpkgs` technique), or
2. modify the source code -- in a way that arguably would make it less pretty

Are we clear we want to pick the second option?

**Comment by domenkozar** [comment]
I'm also fine with (1) since it means less changes on the long run :shipit:

**Comment by deepfire** [comment]
@domenkozar, done:

```
deepfire@andromedae:~/src/iohk$ nsh
these derivations will be built:
  /nix/store/vlqqvclb5fy9p26hl3k9qk4nk50k4lsh-iohk-ops-0.1.0.0.drv
  /nix/store/4mgdcp6z8qkmy9v0d480mxkl7xz5vn16-ghc-8.0.2-with-packages.drv
building path(s) ‘/nix/store/14f9smc8rfbraz8skdik835wgfvia562-iohk-ops-0.1.0.0’
building path(s) ‘/nix/store/kz4dk3mphcykcw9z5ypkq8iws5w9r7x3-ghc-8.0.2-with-packages’
NIX_PATH set to nixpkgs=/nix/store/vi2bqlxvb1nj953qhm3jgnxnmgl9jwbh-nixpkgs-9b948ea439ddbaa26740ce35543e7e35d2aa6d18-src

[nix-shell:~/src/iohk]$ which iohk-ops
/nix/store/kz4dk3mphcykcw9z5ypkq8iws5w9r7x3-ghc-8.0.2-with-packages/bin/iohk-ops

[nix-shell:~/src/iohk]$ cat /nix/store/kz4dk3mphcykcw9z5ypkq8iws5w9r7x3-ghc-8.0.2-with-packages/bin/iohk-ops
#! /nix/store/fi3mbd2ml4pbgzyasrlnp0wyy6qi48fh-bash-4.4-p5/bin/bash -e
export PATH=/nix/store/jaa2qzigjfppwis1wllidns0yy85d0ms-nix-prefetch-scripts/bin${PATH:+:}$PATH
exec -a "$0" "/nix/store/14f9smc8rfbraz8skdik835wgfvia562-iohk-ops-0.1.0.0/bin/.iohk-ops-wrapped"  "${extraFlagsArray[@]}" "$@"
```

File: default.nix [hunk 1]
```diff
@@ -58,6 +58,14 @@ let
...
     # extra packages to expose, that have no relation to pkgs/default.nix
     stack2nix = compiler.callPackage ./pkgs/stack2nix.nix {};
-    iohk-ops =  compiler.callCabal2nix "iohk-ops" ./iohk {};
+    iohk-ops = pkgs.haskell.lib.overrideCabal
+               (compiler.callCabal2nix "iohk-ops" ./iohk {})
+               (drv: {
+                  executableToolDepends = [ pkgs.makeWrapper ];
+                  postInstall = ''
+                    wrapProgram $out/bin/iohk-ops \
+                    --prefix PATH : "${pkgs.lib.makeBinPath [ pkgs.nix-prefetch-scripts ]}"
```
Comment (domenkozar) [inline default.nix]: Shouldn't this also include `git`?

Comment (deepfire) [inline default.nix]: Done.

**Pull request merged**